src/phoenix/datasets/experiments.py

```diff
@@ -170,3 +214,133 @@ async def async_run_experiment(example: ExampleProtocol) -> ExperimentPayload:
     return Experiment(
         id=experiment_id, dataset_id=dataset.id, dataset_version_id=dataset.version_id
     )
+
+
+def evaluate_experiment(
+    experiment: Experiment,
+    evaluator: ExperimentEvaluator,
+    name: Optional[str] = None,
+    label: Optional[str] = None,
+) -> None:
+    # define wrapper classes to coerce JSON payloads to conform to the input protocols until
+    # we flesh out our clientside models
+
+    @dataclass
+    class ExampleWrapper:
+        id: str
+        input: JSONSerializable
+        output: JSONSerializable
+
+    @dataclass
+    class RunWrapper:
+        id: str
+        output: JSONSerializable
+
+    client = _phoenix_client()
+
+    experiment_id = experiment.id
+    dataset_id = experiment.dataset_id
+    dataset_version_id = experiment.dataset_version_id
+
+    dataset_examples = client.get(
+        f"/v1/datasets/{dataset_id}/examples",
+        params={"version-id": str(dataset_version_id)},
+    ).json()
+
+    experiment_runs = client.get(f"/v1/experiments/{experiment_id}/runs").json()
+
+    # not all dataset examples have associated experiment runs, so we need to pair them up
+    # based on the example id, this assumes that the examples and experiment runs are sorted by id
+    example_run_pairs = []
+    example_cursor, run_cursor = 0, 0
+    while example_cursor < len(dataset_examples) and run_cursor < len(experiment_runs):
+        example = dataset_examples[example_cursor]
+        run = experiment_runs[run_cursor]
+        if example["id"] == run["dataset_example_id"]:
+            wrapped_example = ExampleWrapper(
+                id=example["id"], input=example["input"], output=example["output"]
+            )
+            wrapped_run = RunWrapper(id=run["id"], output=run["output"])
+            example_run_pairs.append((wrapped_example, wrapped_run))
+            example_cursor += 1
+            run_cursor += 1
+        elif example["id"] < run["dataset_example_id"]:
+            example_cursor += 1
+        else:
+            run_cursor += 1
```

Comment:
I think we need to support repetitions for the runs.

Reply:
I agree we need to double check, but I'd prefer to wait until we finalize what the DB support for repetitions looks like first before writing a preemptive change. however, if there's multiple runs with a matching example, won't we just add each example-run pair that matches to the matched set?

Reply:
Okay. I think it's a one-line change shown below.